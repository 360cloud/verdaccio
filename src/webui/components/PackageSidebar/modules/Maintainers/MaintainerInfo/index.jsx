import React from 'react';
import PropTypes from 'prop-types';

import classes from './style.scss';

export default function MaintainerInfo({title, name, avatar}) {
  let avatarDescription = `${title} ${name}'s avatar`;
  return (
    <div className={classes.maintainer} title={name}>
      <img alt={avatarDescription} src={avatar} title={avatarDescription} />
      <span className="maintainer-name">{name}</span>
    </div>
  );
}
MaintainerInfo.propTypes = {
  title: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  avatar: PropTypes.string.isRequired,
};
