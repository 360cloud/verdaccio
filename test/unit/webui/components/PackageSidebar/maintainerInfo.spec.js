/**
 * MaintainerInfo component
 */

import React from 'react';
import { shallow } from 'enzyme';
import MaintainerInfo from '../../../../../src/webui/components/PackageSidebar/modules/Maintainers/MaintainerInfo/index';

console.error = jest.fn();

describe('<PackageSidebar /> : <Maintainers /> <MaintainerInfo />', () => {
  test('should throw error for required props', () => {
    shallow(<MaintainerInfo />);
    expect(console.error).toHaveBeenCalled();
  });

  test('should load the component and match with snapshot', () => {
    const wrapper = shallow(<MaintainerInfo avatar={'http://xyz.com/profile.jpg'} name={'test'} title={'test-title'} />);
    expect(wrapper.find('.maintainer').prop('title')).toEqual('test');
    expect(wrapper.find('img').prop('src')).toEqual(
      'http://xyz.com/profile.jpg'
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
