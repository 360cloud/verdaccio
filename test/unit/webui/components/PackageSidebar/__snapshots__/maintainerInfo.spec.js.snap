// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PackageSidebar /> : <Maintainers /> <MaintainerInfo /> should load the component and match with snapshot 1`] = `"<div class=\\"maintainer\\" title=\\"test\\"><img alt=\\"test-title test&#x27;s avatar\\" src=\\"http://xyz.com/profile.jpg\\" title=\\"test-title test&#x27;s avatar\\"/><span class=\\"maintainer-name\\">test</span></div>"`;
